src/scripts/daterangepicker/calendar-header-view.coffee

```diff
@@ -54,11 +54,21 @@ class CalendarHeaderView
     {'arrow-hidden': !@currentDate.isWithinBoundaries(date)}
 
   nextArrowCss: ->
-    [cols, rows] = @period.dimentions()
-    date = @firstDate().clone().add(cols * rows, @period())
-    {'arrow-hidden': !@currentDate.isWithinBoundaries(date)}
-
-
+    currentDate = @currentDate().clone()
+
+    # Calculate the _first_ available date in the next period
+    nextPeriodDate =
+      # Day and week calendars are displayed 'monthly' (6 weeks)
+      if @period() in ['day', 'week']
+        currentDate.endOf('month').add(1, 'day')
+      # Month and quarter calendars are displayed yearly
+      else if @period() in ['month', 'quarter']
+        currentDate.endOf('year').add(1, 'day')
+      # Year calendar is displayed in 10 year chunks
+      else if @period() == 'year'
+        @firstYearOfDecade(currentDate).add(10, 'year')
+
+    {'arrow-hidden': !@currentDate.isWithinBoundaries(nextPeriodDate)}
```

Comment:
I think this can be simplified significantly. Year, Quarter & Month work fine as is. So we can just default to the old implementation in those cases:
```coffeescript
date = @firstDate().clone().add(cols * rows, @period())
```
for day & week, however, we should be looking at the beginning of the month rather than whatever `.add(cols * rows, @period()` gives us.
So, I think something like this should work:
```coffeescript
nextArrowCss: ->
  [cols, rows] = @period.dimentions()
  date = @firstDate().clone().add(cols * rows, @period())
  date = date.startOf('month') if @period() in ['day', 'week']
  {'arrow-hidden': !@currentDate.isWithinBoundaries(date)}
```